src/liboslexec/batched_llvm_gen.cpp

```diff
@@ -4313,19 +4313,15 @@ LLVMGEN (llvm_gen_raytype)
     llvm::Value * sg = rop.sg_void_ptr();
     if (Name.is_constant()) {
         // We can statically determine the bit pattern
-        ustring name = ((ustring *)Name.data())[0];
-        llvm::Value *args[] = {
-            sg,
-            rop.ll.constant (rop.shadingsys().raytype_bit (name))};
-
-        llvm::Value *ret = rop.ll.call_function (rop.build_name("raytype_bit"), args);
-
-        if(!result_is_uniform)
-        {
+        llvm::Value* bit;
+        bit = rop.ll.constant(rop.shadingsys().raytype_bit(Name.get_string()));
+        llvm::Value* raytype = rop.ll.op_load_int(
+            rop.llvm_global_symbol_ptr(Strings::raytype));
+        llvm::Value* bitanded = rop.ll.op_and(raytype, bit);
+        llvm::Value* ret = rop.ll.op_int_to_bool_to_int(bitanded);
+        if (!result_is_uniform)
             ret = rop.ll.widen_value(ret);
-        }
         rop.llvm_store_value (ret, Result);
-
     } else {
         FuncSpec func_spec("raytype_name");
         // No way to know which name is being asked for
```

Comment:
For the non-const case suggest following the same pattern you did for llvm_gen, go into wide_shading.cpp and rename the uniform and varying versions of raytype_name to raytype_bit.

For the uniform verson of raytype_name just return the bit by removing the  bitwise AND `return (bsg->uniform.raytype & bit)`, and in the batched_llvm_gen add a bitwise AND to the raytype from batchedShaderGlobals (very similar to llvm_gen version).

For the varying version, leave the foreach_unique loop, but again drop the bitwise AND `int ray_is_named_type = ((bsg->uniform.raytype & bit) != 0)` and just pass the `bit` variable into `assign_all`.  But back in the batched_llvm_gen you will now need to allocate a varying temporary int to pass to varying version of raytype_bit.  IE:
```
BatchedBackendLLVM::TempScope temp_scope(rop);

// We need wide place to hold the results of raytype_bit 
llvm::Value * loc_raytype_bit = rop.getOrAllocateTemp (TypeSpec(TypeDesc::INT), false /*derivs*/, false /*is_uniform*/, false /*forceBool*/, std::string("raytype_bit"));
            
llvm::Value *args[] = {
                sg,
                rop.llvm_void_ptr (loc_raytype_bit ),
                rop.llvm_void_ptr (Name),
                rop.ll.mask_as_int(rop.ll.current_mask())};
rop.ll.call_function (rop.build_name(func_spec), args);

llvm::Value* raytype= rop.ll.op_load(loc_raytype_bit );
llvm::Value* wide_bit = rop.ll.widen_value(bit);
llvm::Value* bitanded = rop.ll.op_and(raytype, wide_bit );
llvm::Value* ret = rop.ll.op_int_to_bool_to_int(bitanded);
OSL_DASSERT(!result_is_uniform);
rop.llvm_store_value (ret, Result);
```

To simplify, you can can probably combine some of the code between the uniform and non-uniform branches